docs/source/performance/cudagraphs.md

```diff
@@ -0,0 +1,26 @@
+# Cudagraphs for NeMo
+
+###  Background and Motivation
+Cudagraphs are a performance feature used mainly to eliminate reduce the presence of GPU idling due to host overhead and jitter. Commonly, if the host cannot issue kernels fast enough, the GPU may empty its work queue and begin idling. Cudagraphs solve this by encapsulating a sequence of many kernels into single unit. Thus, the host can launch multiple kernels by calling a single cudagraph, which greatly reduces the amount of work required by the host in comparison to eagerly launching each kernel individually.
+The current cudagraph implementation in NeMo maps each transformer layer’s forward and backpasses to a cudagraph. On the first step, `CudaGraphManager` intercepts the forward pass to each transformer layer and uses it to trace the transformer layer via stream capture. On subsequent steps, `CudagraphManager`, rather than of executing the layer eagerly, reroutes calls into a single cudagraph, 
```

Comment:
`The current cudagraph implementation in NeMo maps each transformer layer’s forward and backpasses to a cudagraph.`
> `The current implementation in NeMo captures all the kernels per Transformer layer in each of the forward and backward passes as a single graph.`